ccips/ccip-020/ccip-020-graceful-protocol-shutdown.md

```diff
@@ -0,0 +1,108 @@
+
+In light of these challenges and after careful consideration of community feedback, this CCIP proposes a graceful shutdown of the CityCoins Protocol. The shutdown process will involve disabling the mining and stacking functionality while ensuring that MIA and NYC token holders are fairly compensated through a redemption mechanism utilizing the STX held in the city treasuries.
+
+The primary objectives of this CCIP are to:
+
+- Protect the interests of MIA and NYC token holders
+- Honor the milestones achieved by the CityCoins Protocol
+- Honor the gift agreement with the City of Miami
+- Facilitate a smooth transition and minimize disruption to the ecosystem
+
+## Specification
+
+### Disable Mining
+
+This CCIP will disable mining in the current mining contract, such that:
+
+- Mining will stop at the block height the shutdown occurs
+- Any mining transactions sent after the shutdown block height will fail
+- Any mining claims made in this and prior contracts will still function
+
+Mining claims are based on block heights and should be routed to the correct contracts:
+
+| Contract                  | Start Block | End Block |
+| ------------------------- | ----------- | --------- |
+| miamicoin-core-v1         | 24497       | 58917     |
+| newyorkcitycoin-core-v1   | 37449       | 58922     |
+| miamicoin-core-v2         | 58921       | 96779     |
+| newyorkcitycoin-core-v2   | 58925       | 96779     |
+| ccd006-citycoin-mining    | 96779       | 107389    |
+| ccd006-citycoin-mining-v2 | 107389      | TBD       |
+
+### Disable Stacking
+
+This CCIP will disable stacking in the current stacking contract, such that:
+
+- Stacking will stop at the block height the shutdown occurs
+- Any stacking transactions sent after the shutdown block height will fail
+- Any stacking claims made in this and prior contracts will still function
+- Any locked CityCoins can be claimed from the future cycle(s)
+
+For example, at the time of writing this we are in cycle 80. If a user is stacking from cycles 77-84 then the following would be true:
+
+| Cycle | Claim STX | Claim CityCoins | Description               |
+| ----- | --------- | --------------- | ------------------------- |
+| 77    | Yes       | No              | same as before            |
+| 78    | Yes       | No              | same as before            |
+| 79    | Yes       | No              | same as before            |
+| 80    | Yes       | No              | same as before            |
+| 81    | No        | No              | shutdown, no STX to claim |
+| 82    | No        | No              | shutdown, no STX to claim |
+| 83    | No        | No              | shutdown, no STX to claim |
+| 84    | No        | Yes             | retrieve locked CityCoins |
+
+### Redemption Extensions
+
+This CCIP will implement two new redemption extensions, one each for MIA and NYC.
```

Comment:
In order to honor the gift agreement of MIA, there should be a single extension that is used by both MIA and NYC holders. One single pool of STX.
The pool can be filled by future STX stacking rewards until the claim period ends.

Reply:
My understanding is that NYC Treasury has nothing to do with the MIA gift agreement.
Even if we liquidate, NYC holders will swap to STX in NYC Treasury (total amount of STX swaps in NYC holders <= total amount of STX in NYC Treasury), but MIA holders still need to confirm the total amount of swaps due to the gift agreement. 

Is there any reason why they should be viewed as one single pool? I am not convinced.

Reply:
I'm thinking that in case of liquidation, STX swap should be done separately for each citicoin.

Because I think NYC holders have rights only in STX within NYC's STX TRESURY, and MIA holders have rights only in STX within MIA's STX TRESURY.

Reply:
If I'm missing a point or not understanding it well, please give me feedback.  :)

Reply:
Both NYC and MIA holders don't have any rights on any STX of the treasury.

If we decide to take out stx from one or the other treasury to compensate CC holders, this is a decision on the DAO level, not the city level.

Reply:
Addressed [in this comment](https://github.com/citycoins/governance/pull/30#issuecomment-2018791406)

I think they should be separate.